core/src/main/scala/dagr/core/execsystem/TaskManager.scala

```diff
@@ -327,20 +328,21 @@ class TaskManager(taskManagerResources: SystemResources = TaskManagerDefaults.de
     * @return for each completed task, a map from a task identifier to a tuple of the exit code and the status of the `onComplete` method
     */
   private def updateCompletedTasks(): Map[TaskId, (Int, Boolean)] = {
-    val completedTasks: Map[TaskId, (Int, Boolean)] = taskExecutionRunner.completedTasks()
-    val emptyTasks = graphNodesInStateFor(GraphNodeState.NO_PREDECESSORS).filter(_.task.isInstanceOf[Task.EmptyTask])
-
-    emptyTasks.foreach { node =>
+    // Get the tasks that are not eligible to be scheduled/executed, and have no predecessors (can be completed).
+    val toCompleteTasks = graphNodesInStateFor(GraphNodeState.NO_PREDECESSORS).filterNot(_.task.isInstanceOf[UnitTask])
```

Comment:
This is confusing to me ... wouldn't this also grab e.g. `Pipeline` instances that have no predecessors and complete them?  If not, can you add to the comment as to how this doesn't select those?